### PR TITLE
Remove donations code

### PR DESCRIPTION
**Pull request by sarveshpro**
### To test -
- Do we need Legacy Donations - `donation-payment/[code]` it gets removed in this PR
- Share Functionality was moved out of the donations folder
- `/payment-status` -  functionality restored
- `/claim/[code]` - functionality restored

### CODE REVIEW

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**/claim/\[type\]/\[code\]**](https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/claim/[type]/[code]) | [**/donation-payment/\[code\]**](https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/donation-payment/[code]) |
|:-:|:-:|
|<a href="https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/claim/[type]/[code]">Screenshot of /claim/[type]/[code]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/sI10yDDFPvufdLl3VrvZ_TyRO2m0GXM2sYBQpb2_zhnibsCrShh8QNi7tRdPBCkL9PHdYmJokWowlGiSzCrcFw.png">(view full size)</a> |<a href="https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/donation-payment/[code]">Screenshot of /donation-payment/[code]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Zu3XOp1oQn5KK4SxduhGSDnmsdeY3bxdGdmnsNdjEVXa8iWZ0BcyR2FEQt39BuYvBinZqp5V7JYk9L1wOVCw3A.png">(view full size)</a> |



| [**/**](https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/) | [**/payment-status**](https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/payment-status) |
|:-:|:-:|
|<a href="https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/kBJlgzpTmYtB7gVUDBNehGMUBfBCADJbgr0OhHDjlF8FKcIwpg-JyocvYsPl6oQnZROD3KLsCZjm2ab21pnU6g.png">(view full size)</a> |<a href="https://planet-webapp-mvs3l7f4v-planetapp.vercel.app/payment-status">Screenshot of /payment-status</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/CjPfTPqr22aDrT4uPB2-xxZzGhvt2v9CfIf29HwQjLBxOHy9CMsmzz4wLavcDV9NCN5PsYJQVo-f7LcfECcGTg.png">(view full size)</a> |







Commit 613dd2e976e071bc541c75f738de395fad6f48f1 (https://planet-webapp-mvs3l7f4v-planetapp.vercel.app).

**Review by sagararyal — DISMISSED** [review]
remove all legacy donations code and remove packages.

**Comment by sarveshpro** [comment]
> remove all legacy donations code and remove packages.

Done

**Review by ankitecd — CHANGES_REQUESTED** [review]
> Did `npx depcheck` and found the below unused packages.
> Can you check and resolve the same?
> 

Deleted most of them, unfortunately, some of them I had to restore because they are needed `depcheck` couldn’t determine their dependency.

**Comment by sagararyal** [comment]
@sarveshpro can we also remove unused translation resources.

**Comment by norbertschuler** [comment]
> @sarveshpro can we also remove unused translation resources.

Due to the nature of JavaScript I think there is no tool to check for unused resources (also some resource name are defined dynamically). So we would need to check them manually which probably is quite a time consuming task to do.

File: pages/payment-status.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { useRouter } from 'next/router'
```
Comment (norbertschuler) [inline pages/payment-status.tsx:1]: I am not sure if this page is still used?

**Review by norbertschuler — APPROVED** [review]
I tested redeeming a code and it still works. Also the other parts of the website I click through still seem to work, but I am probably do not covered 100% with my manual tests.

**Comment by norbertschuler** [comment]
> * Do we need Legacy Donations - `donation-payment/[code]` it gets removed in this PR

This was for older versions of the Treecounter-App, but hopefully nobody is using these old versions any more. I think we do not have to support them any more.

**Review by ankitecd — APPROVED** [review]
Looks good

**Pull request merged**